core/services/flux_monitor_test.go

```diff
@@ -216,43 +233,44 @@ func TestPollingDeviationChecker_StartStop(t *testing.T) {
 		default:
 		}
 	})
-	// can be called an arbitrary # of times
 	defer fetcher.AssertExpectations(t)
```

Comment:
Not sure I like the mix of defer `AssertExpectations` and literal `AssertExpectations` think it makes the test less clear and also results in a weird stack trace when it fails. i.e. it will say the test failed on the final brace of this function, when it would have been clearer if the stack trace pointed to the literal assert.